lib/src/map/map_builder.dart

```diff
@@ -53,6 +44,20 @@ class MapBuilder<K, V> {
     updates(this);
   }
 
+  /// Replaces all elements with elements from a [Map] or [BuiltMap].
+  void replace(map) {
+    if (map is BuiltMap<K, V>) {
+      _replaceFromBuiltMap(map);
+    } else if (map is BuiltMap) {
+      _replaceFromSafeMap(new Map<K, V>.from(map._map));
```

Comment:
If the map already has a safe map, we don't need to copy it, right?
Would it make sense to have a safeMap getter that knows whether to return a copy or not?